test/e2e/scenarios/pod_selector.go

```diff
@@ -2,24 +2,106 @@ package scenarios
+}
+
+var all_pod_selector_np = &networkingv1.NetworkPolicy {
+	ObjectMeta: metav1.ObjectMeta {
+		Name: "all-pods-same-ns-different-labels",
+		Namespace: "boring",
+		ClusterName: string(framework.ClusterA),
+	},
+	Spec: networkingv1.NetworkPolicySpec {
+		PodSelector: metav1.LabelSelector {
+			// all pods on cluster A are selected
+		},
+		PolicyTypes: []networkingv1.PolicyType{networkingv1.PolicyTypeIngress},
+		Ingress: []networkingv1.NetworkPolicyIngressRule {{
+			From: []networkingv1.NetworkPolicyPeer {{
+				PodSelector: &metav1.LabelSelector {
+					MatchLabels: map[string]string {
+						"role": "frontend",
+					},
+				},
+			}},
+			Ports: []networkingv1.NetworkPolicyPort {{
+				Port: &intstr.IntOrString{Type: intstr.Int, IntVal: 1234},
+			}},
+		}},
+	},
+}
+
 var _ = PDescribe("[Dataplane] PodSelector based Network Policies for ingress between clusters", func() {
 	PContext("Allow communication between selected pods on different clusters in same project", func() {
 		PIt("Should allow communication between selected pods", func() {
 			By("creating listener pod 1 with label 1 in cluster 1 in namespace 1")
+			// create pod1 with label "role = backend" in "boring namespace"
 			By("creating connecting pod 2 with label 2 in cluster 2 in namespace 1")
+			// create pod2 with label "role = frontend" in "boring namespace"
 			By("creating network policy in cluster 1 that allows communication to pod 1 from pod 2 in cluster 2")
+			_,err := f.ClusterClients[framework.ClusterA].NetworkingV1().NetworkPolicies(f.Namespace).Create(pod_selector_np)
+			framework.ExpectNoError(err)
 			By("testing connectivity between the two pods")
+			// connectivity between pod 1 and pod2
+			// Negative tests
+			By("creating connecting pod 3 with label 3 in cluster 2 in namespace 1")
+			// create pod 3 with label "role = another" in "boring namespace"
+			By("testing non connectivity between the two pods")
+			// Tests no connectivity between pod 3 and pod 1
 		})
 	})
 
-	PContext("[Dataplane] Allow communication between selected pods on first cluster to all pods in second cluster in same project", func() {
+	PContext("[Dataplane] Allow communication between all pods on first cluster to selected pods in second cluster in same project", func() {
 		PIt("Should allow communication between selected pods on 1 cluster and all pods on other cluster", func() {
 			By("creating listener pod 1 with label 1 in cluster 1 in namespace 1")
+			// create pod1 with label "role = backend" in "boring namespace"
 			By("creating connecting pod 2 with label 2 in cluster 2 in namespace 1")
+			// create pod2 with label "role = frontend" in "boring namespace"
 			By("creating network policy in cluster 1 that allows communication to pod 1 from all pods in cluster 2")
+			_, err := f.ClusterClients[framework.ClusterA].NetworkingV1().NetworkPolicies(f.Namespace).Create(all_pod_selector_np)
+			framework.ExpectNoError(err)
+			By("testing connectivity between the two pods")
```

Comment:
with the refactored framework some of those steps will be just one, because you just invoke the check, but looks good.

Reply:
the framework has pod to service connection check. We need to check pod to pod connection here. Or pod to service should suffice?